### DISCUSSION TITLE
Lead-acid battery parameter sets

### DISCUSSION THREAD

**Discussion by wranson22** [pybamm-team/PyBaMM#2882] [General]
Hi there, I'm new to pybamm, and am using it for a final design project at University.
I am looking to simulate a lead acid battery, and am running into an issue with Parameter sets. 
my scenario is a PV system with a set of solar panels that charges the battery, and a load that draws power from either the batteries or the Solar panels
there's only one default parameter set for Lead-Acid batteries, (Sulzer2019) and it's for a battery with different specs than the ones i'm using, so i'm trying to change some parameters to get closer to my battery's specs but there is really a lot of guess work while I try to get the code running. 

```
    parameter_values = pybamm.ParameterValues(chemistry = pybamm.parameter_sets.Sulzer2019) # Define parameters, Sulzer2019 is a built in parameter set for Lead-Acid batteries
    parameter_values["Cell capacity [A.h]"] = 155.0  #capacity listed on batteries
    parameter_values["Cell cooling surface area [m2]"] = 0.16  
    parameter_values["Cell volume [m3]"] = 0.005
    parameter_values["Current function [A]"] = 150
    parameter_values["Electrode height [m]"] = 0.250
    parameter_values["Electrode width [m]"] = 0.15
    parameter_values["Lower voltage cut-off [V]"] = 10   #if the 12V lead acid battery reads below 10V, it's likely close to dead
    parameter_values["Negative electrode thickness [m]"] = 0.001
    parameter_values["Positive electrode pore size [m]"] = 1e-05
    parameter_values["Upper voltage cut-off [V]"] = 14   #if it reads above 14 V, it's fully charged.
    parameter_values.process_model(model)

    print(parameter_values)

    sim = pybamm.Simulation(model, parameter_values=parameter_values, experiment=experiment, solver=pybamm.CasadiSolver())
    sim.solve([0,3600]) # solve for 1 hour
    sim.plot()
    simulate_battery(300, 600, 1, 155, 12, 2, 1)    # power_draw < charge_power
```

The error I'm getting: 
- File: "C:\ ... \pybamm\solvers\base_solver.py", line 491, in calculate_consistent_state
   raise pybamm.SolverError(pybamm.expression_tree.exceptions.SolverError: Could not find consistent states: Could not find acceptable solution: .../casadi/core/rootfinder.cpp:279: rootfinder process failed. Set 'error_on_fail' option to false to ignore this error.

This code will run with the Sulzer2019 parameters, but it runs into errors when I start changing parameter values

Batteries I'm using: https://www.vmaxtanks.com/SLR155-AGM-Solar-Battery-_p_66.html (very little info available from Manufacturer)

Questions: 
- what other parameters should I look at changing in the Sulzer 2019 set? 
- How can I get my guesses more accurate?
- Any recommendations for troubleshooting this error and figuring out how to get my code running? 

code attached below:
[battery_code_041823.txt](https://github.com/pybamm-team/PyBaMM/files/11266663/battery_code_041823.txt)

**Comment by valentinsulzer** [comment]
Make sure your parameters are for one single 2V cell, not the whole 12V battery (6 cells in series). This applies to the voltage limits. Maybe also applies to capacity if you have some parallel connections. Given your lack of info about the cell I would just multiply "electrode height" by "desired capacity"/"Sulzer2019 capacity". This should get you most of the way there

**Reply by wranson22** [reply]
Thank you for the swift response! I misunderstood that it was parameters for the individual cells. looking at it now, the Sulzer2019 model is likely a lot closer of a model than I originally thought. I see in the parameters, the number of cells in series to make a battery is 6 and the number of electrodes in parallel to make a cell is 8. 
Now the issue I'm having is it hits the limits for the terminal voltage, both when I'm charging and discharging. Is 300 Watts for a whole hour too much? does that mean and how can I proceed
[battery_code_041823_2.txt](https://github.com/pybamm-team/PyBaMM/files/11267377/battery_code_041823_2.txt)

**Comment by yazyr** [comment]
I'm facing the same issue as yours @wranson22. I'm getting an IDASolver Error saying Minimum voltage or Maximum voltage is triggered, even when charging and discharging until the cut-off voltages. Would be nice if someone clarifies this through
